streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -59,6 +60,12 @@
 
     private final static int UNKNOWN = -1;
     public final static int NOT_AVAILABLE = -2;
+    private final static int VERSION_ONE = 1;
+    private final static int VERSION_TWO = 2;
+    private final static int VERSION_THREE = 3;
+    private final static int EARLIEST_PROBEABLE_VERSION = VERSION_THREE;
+    protected int minUserMetadataVersion = UNKNOWN;
+    protected Set<Integer> supportedVersions = new HashSet<>();
```

Comment:
This field is only used for testing purposes?

Reply:
For now, yes. However, after we bump metadata version to 4, parts of the logic of `FutureStreamsPartitionAssignor` will move into `StreamsPartitionAssignor` and than also `StreamsPartitionAssignor` need this.

Reply:
Ack.